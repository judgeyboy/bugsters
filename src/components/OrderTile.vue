<template>
  <div class="tile bugster-tile">
    <div class="tile-icon bugster-tile-icon">
      <div class="example-tile-icon">
        <figure class="avatar bugster-avatar">
          <img :src="'http://' + order.site + '/favicon.ico'" alt="Avatar">
          <i class="avatar-presence online"></i>
        </figure>
      </div>
    </div>
    <div class="tile-content">
      <p class="tile-title">{{ order.site }}</p>

      <span class="badge badge-passed" :data-badge="order.passed"></span>

      <span class="badge badge-partial" :data-badge="order.partial"></span>

      <span class="badge badge-failed" :data-badge="order.failed"></span>

      <span class="badge badge-open" :data-badge="order.open"></span>

      <div class="bar bar-sm">
        <div class="bar-item tooltip" :data-tooltip="order.percentage + '%'" :style="{width:order.percentage + '%'}"></div>
      </div>
    </div>
    <div class="tile-action">
      <router-link class="btn btn-link" :to="'orders/' + order.id">
        <i class="icon icon-arrow-right"></i>
      </router-link>
    </div>
  </div>
</template>

<script>
export default {
  name: 'OrderTile',
  props: ['order']
}
</script>

<style lang="sass" scoped>
.bugster-avatar
  background: transparent

.bugster-tile
  margin: 1.5rem 0

  &:first-of-type
    margin: .4rem 0

.bugster-tile-icon
  padding-right: .4rem

.badge-passed::after
  background-color: green

.badge-failed::after
  background-color: red

.badge-partial::after
  background-color: orange

.badge-open::after
  background-color: #50596c
</style>
