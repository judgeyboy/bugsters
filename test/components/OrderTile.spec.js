import {shallow} from 'vue-test-utils'

import OrderTile from '@/components/OrderTile'

describe('OrderTile component', () => {
  it('should be a vue instance', () => {
    const wrapper = shallow(OrderTile, {
      propsData: {
        order: {
          id: 'kejkelge',
          state: 'in progress'
        }
      }
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(OrderTile, {
      propsData: {
        order: {
          id: 'kejkelge',
          state: 'in progress'
        }
      }
    })
    const $html = wrapper.vm.$el.outerHTML
    expect($html).toMatchSnapshot()
  })
})
