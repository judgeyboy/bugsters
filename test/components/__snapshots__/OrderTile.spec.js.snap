// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderTile component should match snapshot 1`] = `

<div class="tile bugster-tile">
  <div class="tile-icon bugster-tile-icon">
    <div class="example-tile-icon">
      <figure class="avatar bugster-avatar">
        <img src="http://undefined/favicon.ico"
             alt="Avatar"
        >
        <i class="avatar-presence online">
        </i>
      </figure>
    </div>
  </div>
  <div class="tile-content">
    <p class="tile-title">
    </p>
    <span class="badge badge-passed">
    </span>
    <span class="badge badge-partial">
    </span>
    <span class="badge badge-failed">
    </span>
    <span class="badge badge-open">
    </span>
    <div class="bar bar-sm">
      <div data-tooltip="undefined%"
           class="bar-item tooltip"
      >
      </div>
    </div>
  </div>
  <div class="tile-action">
    <router-link to="orders/kejkelge"
                 class="btn btn-link"
    >
      <i class="icon icon-arrow-right">
      </i>
    </router-link>
  </div>
</div>

`;
